src/main/core/manager.rs

```diff
@@ -230,7 +232,20 @@ impl<'a> Manager<'a> {
                     .try_into()
                     .unwrap()
             })
-            .get();
+            .get()
+            .try_into()
+            .unwrap();
+
+        let parallelism: usize = self
+            .config
+            .general
+            .parallelism
+            .unwrap()
+            .get()
+            .try_into()
+            .unwrap();
+
+        let num_threads = std::cmp::min(num_workers, parallelism);
```

Comment:
It turns out that I wasn't using the `num_workers` option correctly, which changed the behaviour of that configuration option. Since the new scheduler doesn't support multiple hosts per thread, I've removed the `num_workers` option instead and removed this line.